index.bs

```diff
@@ -900,77 +900,102 @@ See [[#naming-is-hard]] for general (cross-language) advice on naming.
-developers. Instead, they should <a>queue a task</a> to modify author-observable state (such as an
-object property).
-
-<h3 id="js-gc">Do not expose garbage collection</h3>
-
-Web APIs should not expose a way for author code to deduce when/if garbage collection of JavaScript
-objects has run.
-
-Although some APIs may expose garbage collection, such as some implementations of
-{{Document/getElementsByTagName}}, and the JavaScript <a
-href="https://github.com/tc39/proposal-weakrefs">WeakRefs proposal</a> (which has multiple
-implementer support), API designers are strongly encouraged to avoid points in their own APIs that
-depend on this timing.
-
-The reason for this is somewhat subtle. The more that Web API semantics are affected by garbage
-collection timing (or whether objects are collected at all), the more programs will be affected by
-changes in this timing. But user agents differ significantly in both the timing of garbage
-collection and whether certain objects are collected at all, which means the resulting code will be
-non-interoperable. Worse, according to the usual rules of game theory as applied to browsers, this
-kind of scenario could force other user agents to copy the garbage collection timing of the
-original in order to create interoperability. This would cause current garbage collection
-strategies to ossify, preventing improvement in one of the most dynamic areas of JavaScript
-virtual machine technology.
-
-In particular, this means that you shouldn't expose any API that acts as a weak reference, e.g. with a
-property that becomes <code highlight="js">null</code> once garbage collection runs. Such freeing of
-memory should be entirely deterministic.
-
-For example, currently, {{Document/getElementsByTagName}} permits reusing the {{HTMLCollection}}
-which it creates, when it's called with the same receiver and tag name. In practice, engines reuse
-the output if it has not been garbage collected. This creates behavior which differs based on the
-details of garbage collection, which is strongly discouraged. If {{Document/getElementsByTagName}}
-were designed today, the advice to the specification authors would be to either reliably reuse
-the output, or to produce a new {{HTMLCollection}} each time it's invoked.
-
-The case of {{Document/getElementsByTagName}} is particularly insidious because nothing about
-the API visibly relates to garbage collection details. By contrast, special APIs for this particular
-purpose such as `WeakRef` and `FinalizationGroup` have their GC interaction as a specific part of
-their contract with the developer.
+Don't modify data accessed from JavaScript APIs
+while a JavaScript <a>event loop</a> is running.
+
+A JavaScript Web API is generally a wrapper around
+a feature implemented in a lower-level language,
+such as C++ or Rust.
+Unlike those languages,
+JavaScript guarantees that once a piece of code begins executing,
+it will continue executing until it has completed.
```

Comment:
Does it really?

Reply:
Are there cases when it wouldn't?

Reply:
The wording sounds like Javascript magically makes the underlying program bug-and-failure-free, which is not true.

JavaScript guarantees that once a piece of code begins executing,

to

The general expectation when using JavaScript is that once a piece of code begins executing,

Reply:
Done with a light reword.